apps/language_server/lib/language_server/providers/execute_command/manipulate_pipes/ast.ex

```diff
@@ -0,0 +1,67 @@
+defmodule ElixirLS.LanguageServer.Providers.ExecuteCommand.ManipulatePipes.AST do
+  @moduledoc """
+  AST manipulation helpers for the `ElixirLS.LanguageServer.Providers.ExecuteCommand.ManipulatePipes`\
+  command.
+  """
+
+  @doc "Parses a string and converts the first function call, pre-order depth-first, into a pipe."
+  def to_pipe(code_string) do
+    {piped_ast, _} =
+      code_string
+      |> Code.string_to_quoted!()
```

Comment:
This part of the code breaks upon `\r` characters. I had added a String.replace here to deal with that, but that broke literal strings.

Since we can have arbitrary sigils as well, I'm starting to think that we'll have to just not support `\r` as a line separator.
What do you think about this @lukaszsamson?

Reply:
Good. Let's add a quick check fail fast when \r is detected.